src/librustc_trans/trans/mir/block.rs

```diff
@@ -254,44 +260,46 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
         }
     }
 
-    fn get_personality_slot(&mut self, bcx: Block<'bcx, 'tcx>) -> ValueRef {
+    fn get_personality_slot(&mut self, bcx: &BlockAndBuilder<'bcx, 'tcx>) -> ValueRef {
         let ccx = bcx.ccx();
         if let Some(slot) = self.llpersonalityslot {
             slot
         } else {
             let llretty = Type::struct_(ccx, &[Type::i8p(ccx), Type::i32(ccx)], false);
-            let slot = base::alloca(bcx, llretty, "personalityslot");
+            let slot = bcx.with_block(|bcx| {
+                base::alloca(bcx, llretty, "personalityslot")
+            });
             self.llpersonalityslot = Some(slot);
-            base::call_lifetime_start(bcx, slot);
+            bcx.with_block(|bcx| {
+                base::call_lifetime_start(bcx, slot);
+            });
             slot
         }
     }
 
-    fn make_landing_pad(&mut self, cleanup: Block<'bcx, 'tcx>) -> Block<'bcx, 'tcx> {
+    fn make_landing_pad(&mut self, cleanup: Block<'bcx, 'tcx>) -> BlockAndBuilder<'bcx, 'tcx> {
         let bcx = cleanup.fcx.new_block("cleanup", None);
         // FIXME(#30941) this doesn't handle msvc-style exceptions
         *bcx.lpad.borrow_mut() = Some(LandingPad::gnu());
+        let bcx = bcx.build();
         let ccx = bcx.ccx();
-        let llpersonality = bcx.fcx.eh_personality();
+        let llpersonality = bcx.fcx().eh_personality();
         let llretty = Type::struct_(ccx, &[Type::i8p(ccx), Type::i32(ccx)], false);
-        let llretval = build::LandingPad(bcx, llretty, llpersonality, 1);
-        build::SetCleanup(bcx, llretval);
-        let slot = self.get_personality_slot(bcx);
-        build::Store(bcx, llretval, slot);
-        build::Br(bcx, cleanup.llbb, DebugLoc::None);
+        let llretval = bcx.landing_pad(llretty, llpersonality, 1, bcx.fcx().llfn);
+        bcx.set_cleanup(llretval);
+        let slot = self.get_personality_slot(&bcx);
+        bcx.store(llretval, slot);
+        bcx.br(cleanup.llbb);
         bcx
     }
 
     fn unreachable_block(&mut self) -> Block<'bcx, 'tcx> {
-        match self.unreachable_block {
-            Some(b) => b,
-            None => {
-                let bl = self.fcx.new_block("unreachable", None);
-                build::Unreachable(bl);
-                self.unreachable_block = Some(bl);
-                bl
-            }
-        }
+        self.unreachable_block.unwrap_or_else(|| {
+            let bl = self.fcx.new_block("unreachable", None);
+            build::B(bl).unreachable();
```

Comment:
Why not `bl.build()`? Sure, the `build::B()` will not create a whole new builder (a temporary downside), but I think consistency should be priority.